provisioning/miniprov/test/test_cdf.py

```diff
@@ -52,6 +52,16 @@ def test_disks_empty(self):
             '{ runs_confd = Some True, io_disks = { meta_data = None Text, data = [] : List Text } }',
             str(val))
 
+    def test_pooldesc_empty(self):
```

Comment:
I would suggest one more test case - Mock the confstore in such a way that list of disk refs becomes empty and make sure that Dhall will not complain (i.e. the construction is still valid in terms of Dhall types).

Reply:
Also request you to add a test case where total number of data devices are < data_units + parity_units + spare_units to confirm that this case is handled after adding the validation as metioned above.